mapper/tedge_json/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+//! A library to create the ThinEdgeJson from bytes of json data by Validating it.
+//! It also serialize the ThinEdgeJson data.
+
+use chrono::format::ParseError;
+use chrono::prelude::*;
+use json::JsonValue;
+
+/// ThinEdgeJson is represented in this struct
+/// Since json does not understand DateTime format, the time stamp is represented as a string
+/// Before populating the struct members the thinedge json values and names will be validated
+
+pub struct ThinEdgeJson {
+    pub timestamp: String,
+    pub values: Vec<ThinEdgeValue>,
+}
+
+pub enum ThinEdgeValue {
+    Single(SingleValueMeasurement),
+    Multi(MultiValueMeasurement),
+}
+
+pub struct SingleValueMeasurement {
+    pub name: String,
+    pub value: f64,
+}
+
+pub struct MultiValueMeasurement {
+    pub name: String,
+    pub values: Vec<SingleValueMeasurement>,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct CumulocityJson {
+    c8y_json: JsonValue,
+}
```

Comment:
This is unrelated to `ThinEdgeJson` and MUST stay in `c8y_translator_lib`.

Reply:
yes, moved.